infrastructure/json/src/main/java/tech/pegasys/teku/infrastructure/json/types/DeserializableObjectTypeDefinition.java

```diff
@@ -80,8 +80,21 @@ public TObject deserialize(final JsonParser p) throws IOException {
 
     final List<String> missingRequiredFields =
         deserializableFields.keySet().stream()
+            .filter(key -> !presentFields.contains(key))
             .filter(
-                key -> !presentFields.contains(key) && deserializableFields.get(key).isRequired())
+                key -> {
+                  final DeserializableFieldDefinition<TObject, TBuilder> objectField =
+                      deserializableFields.get(key);
+                  if (objectField.isRequired()) {
+                    return true;
+                  }
+                  // set optional missing fields to Optional.empty()
+                  if (objectField instanceof OptionalDeserializableFieldDefinition) {
+                    ((OptionalDeserializableFieldDefinition<TObject, TBuilder, ?>) objectField)
+                        .setFieldToEmpty(builder);
+                  }
+                  return false;
+                })
```

Comment:
for the sake of readability, can we do this explicitly and intentionally?
It works here, its just obfuscated a bit, and we'd be better off leaving the filter from the original and doing another pass through for all the fields that are optional that aren't present... same result just more easy to digest..
```
deserializableFields.keySet().stream()
     .filter(key -> !presentFields.contains(key) && !deserializableFields.get(key).isRequired())
    .forEach(key -> {
       final DeserializableFieldDefinition<TObject, TBuilder> objectField =
                      deserializableFields.get(key);
      if (objectField instanceof OptionalDeserializableFieldDefinition) {
          ((OptionalDeserializableFieldDefinition<TObject, TBuilder, ?>) objectField)
                        .setFieldToEmpty(builder);
      }
   })
```
 * guarantee thats not working currently, but for demonstration purposes :)
Then the old code dealing with missing required fields can be back to previous, it'll just make it easier to reason about.

Reply:
might even be worth a private fn that `initializeMissingOptioalFields`, and maybe `reportMissingRequiredFields` for future us...

Reply:
maybe we can add a `Optional<OptionalDeserializableFieldDefinition> DeserializableFieldDefinition:: toOptionalDeserializableFieldDefinition` , like other `toVersion` for data structures. Maybe shorter names.. :)

Reply:
Did a small refactor in latest commit, think it's more readable